src/main/java/com/book/booksstore/validator/PasswordMatchesValidator.java

```diff
@@ -0,0 +1,19 @@
+package com.book.booksstore.validator;
+
+import com.book.booksstore.dto.CreateUserRequestDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class PasswordMatchesValidator implements
+        ConstraintValidator<PasswordMatch, CreateUserRequestDto> {
+
+    @Override
+    public boolean isValid(
+            CreateUserRequestDto dto,
+            ConstraintValidatorContext context) {
+        if (dto.getPassword() == null || dto.getRepeatPassword() == null) {
+            return false;
+        }
+        return dto.getPassword().equals(dto.getRepeatPassword());
+    }
```

Comment:
```suggestion
public class FieldsValueMatchValidator
        implements ConstraintValidator<FieldsValueMatch, Object> {
    private String field;
    private String fieldMatch;

    public void initialize(FieldsValueMatch constraintAnnotation) {
        this.field = constraintAnnotation.field();
        this.fieldMatch = constraintAnnotation.fieldMatch();
    }

    public boolean isValid(Object value, ConstraintValidatorContext context) {
        // realization 
    }
}
```
It will be a more scalable solution

Reply:
refactored